### PR TITLE
added catch for fetching single product in cart synchronization

### LINKED ISSUE #3632: cart sync not working
## Current behavior
I have added two products on my local Magento instance while being logged in.
I log into VSF and I don't see the items in the cart.
The items do show up in the `api/cart/pull` response, sometimes.
## Steps to reproduce the issue
Sorry the gif is long, magento takes forever to respond 
cartsync

## Expected behavior
Cart on client should reflect what on server (I think)
## Can you handle fixing this bug by yourself?

- [ ] YES
- [X] NO

## Environment details
- Browser: Chrome Version 77.0.3865.90 (Official Build) (64-bit)
- OS: Windows 10 Pro 1809
- Node: v10.16.3
- Code Version: VSF 1.10.3 / VSF-API 1.10.0

### PR DESCRIPTION
**Pull request by gibkigonzo**
### Related issues

closes #3632

### Short description and why it's useful

This allows to catch errors when user add product to cart in magento that not exist in elasticsearch.

### Which environment this relates to
Check your case. In case of any doubts please read about [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html)

- [ ] Test version (https://test.storefrontcloud.io) - this is a new feature or improvement for Vue Storefront. I've created branch from `develop` branch and want to merge it back to `develop`
- [ ] RC version (https://next.storefrontcloud.io) - this is a stabilisation fix for Release Candidate of Vue Storefront. I've created branch from `release` branch and want to merge it back to `release`
- [x] Stable version (https://demo.storefrontcloud.io) - this is an important fix for current stable version. I've created branch from `hotfix` or `master` branch and want to merge it back to `hotfix`

### Upgrade Notes and Changelog

- [x] No upgrade steps required (100% backward compatibility and no breaking changes)
- [x] I've updated [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing VS sites with this new feature

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and currently important rules acceptance

- [x] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: core/modules/cart/store/actions.ts [hunk 1]
```diff
@@ -703,38 +699,48 @@ const actions: ActionTree<CartState, RootState> = {
...
-              clientCartAddItems.push(
-                new Promise(resolve => {
-                  productActionOptions(serverItem).then((actionOtions) => {
-                    dispatch('product/single', { options: actionOtions, assignDefaultVariant: true, setCurrentProduct: false, selectDefaultVariant: false }, { root: true }).then((product) => {
-                      resolve({ product: product, serverItem: serverItem })
-                    })
-                  })
-                })
-              )
+              const getServerCartItem = new Promise(async resolve => {
```
Comment (andrzejewsky) [inline core/modules/cart/store/actions.ts]: Why are you combining Promises with async/await? Please use just one convention (Currently we prefer async/await). So your solution should looks like this:

```
const getServerCartItem = async () => {
 try {
  const actionOptions = await productActionOptions(serverItem)
 
  if (!actionOptions) {
   return null
  }

  const product = await dispatch('product/single', { options: actionOtions, assignDefaultVariant: true, setCurrentProduct: false, selectDefaultVariant: false }, { root: true })
 
 return { product: product, serverItem: serverItem }
  } catch (err) {
   return null
 }
}
```

Comment (gibkigonzo) [reply core/modules/cart/store/actions.ts]: yeah 😅 I should make all as async/await. Only thing that I have to do then is map those functions in Promise.all
`await Promise.all(clientCartAddItems.map(c => c())`

Comment (gibkigonzo) [reply core/modules/cart/store/actions.ts]: done

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,11 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [1.10.4] - 2019.09.27
+
+### Fixed
+- Added try/catch for fetching single product in cart synchronization - @gibkigonzo (#)
```
Comment (patzick) [inline CHANGELOG.md]:
```suggestion
- Added try/catch for fetching single product in cart synchronization - @gibkigonzo (#3632)
```

**Pull request merged**